thirdparty/CMakeLists.txt

```diff
@@ -74,15 +74,20 @@ macro(build_faiss)
 	set(MMAN_INCLUDE_FLAGS "-I ${KNOWHERE_THIRDPARTY_SRC}/mman-win32")
     endif ()
 
-
+    if(KNOWHERE_GPU_VERSION)
+        set(KNOWHERE_GPU_ENABLE "ON")
+    else()
+        set(KNOWHERE_GPU_ENABLE "OFF")
+    endif()
     set( FAISS_CMAKE_ARGS
             "-DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX}"
             "-DCMAKE_BUILD_TYPE=Release"
             "-DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}"
             "-DCMAKE_CUDA_COMPILER=${CMAKE_CUDA_COMPILER}"
 	    "-DCMAKE_CXX_FLAGS=${CMAKE_CXX_FLAGS} -fpermissive -I ${KNOWHERE_SOURCE_DIR} ${MMAN_INCLUDE_FLAGS}"
             "-DFAISS_ENABLE_PYTHON=OFF"
-            "-DFAISS_ENABLE_GPU=OFF"
+            "-DFAISS_ENABLE_GPU=${KNOWHERE_GPU_ENABLE}"
```

Comment:
"-DFAISS_ENABLE_GPU=${KNOWHERE_GPU_VERSION}"

need not add new variable "KNOWHERE_GPU_ENABLE"

Reply:
> "-DFAISS_ENABLE_GPU=${KNOWHERE_GPU_VERSION}"
> 
> need not add new variable "KNOWHERE_GPU_ENABLE"

Var Type different, KNOWHERE_GPU_ENABLE IS STRING。